app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -1299,6 +1299,11 @@ class BrowserTabFragment :
     }
 
     private fun askSiteLocationPermission(domain: String) {
+        if (!isActiveTab) {
+            Timber.v("Will not launch a dialog for an inactive tab")
+            return
+        }
+
```

Comment:
NIT: I would suggest moving this block of code (and the same 5 ones) to the ViewModel so it can be checked before getting here and tested independently so we don't rely on the fragment for checking it.

Reply:
I think there's a small risk in moving it to the view model that it's more possible when the view model checks, it is the active tab, but there's some small async work happens before the dialog is about to be shown and by then, it's no longer the active tab. Keeping it right at the point of showing the dialog minimises that chance.

Would rather leave them here for now.